src/main/java/com/slack/kaldb/elasticsearchApi/searchRequest/SearchRequest.java

```diff
@@ -0,0 +1,102 @@
+package com.slack.kaldb.elasticsearchApi.searchRequest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import com.slack.kaldb.elasticsearchApi.searchRequest.aggregations.SearchRequestAggregation;
+import com.slack.kaldb.proto.service.KaldbSearch;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SearchRequest {
+
+  private final String index;
+  private final int size;
+  private final String query;
+  private final SearchRequestTimeRange range;
+  private final SearchRequestSort sort;
+  private final List<SearchRequestAggregation> aggregations;
+
+  private SearchRequest(
+      String index,
+      int size,
+      String query,
+      SearchRequestTimeRange range,
+      SearchRequestSort sort,
+      List<SearchRequestAggregation> aggregations) {
+    this.index = index;
+    this.size = size;
+    this.query = query;
+    this.range = range;
+    this.sort = sort;
+    this.aggregations = aggregations;
+  }
+
+  public String getIndex() {
+    return index;
+  }
+
+  public int getSize() {
+    return size;
+  }
+
+  public String getQuery() {
+    return query;
+  }
+
+  public SearchRequestTimeRange getRange() {
+    return range;
+  }
+
+  public SearchRequestSort getSort() {
+    return sort;
+  }
+
+  public List<SearchRequestAggregation> getAggregations() {
+    return aggregations;
+  }
+
+  public KaldbSearch.SearchRequest toKaldbSearchRequest() {
+    return KaldbSearch.SearchRequest.newBuilder()
+        .setIndexName(getIndex())
+        .setQueryString(getQuery())
+        .setStartTimeEpochMs(getRange().getGteEpochMillis())
+        .setEndTimeEpochMs(getRange().getLteEpochMillis())
+        .setHowMany(getSize())
+        .build();
+  }
+
+  public static List<SearchRequest> parse(String postBody) throws JsonProcessingException {
+    ObjectMapper om =
+        new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    // the body contains an NDJSON format, with alternating rows as header/body
+    // @see http://ndjson.org/
+    // @see
+    // https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html#search-multi-search-api-desc
+    List<SearchRequest> requests = new ArrayList<>();
+    for (List<String> pair : Lists.partition(Arrays.asList(postBody.split("\n")), 2)) {
+      JsonNode header = om.readTree(pair.get(0));
+      JsonNode body = om.readTree(pair.get(1));
+
+      // by default grafana formats an empty query as "*" in the request payload, this un-does that
+      String queryString = body.get("query").findValue("query").asText();
+      if (queryString.equals("*")) {
+        queryString = "";
```

Comment:
can we re-write to `*:*` ?

Reply:
Done - had to refactor this anyhow due to a Grafana 7 compatibility issue